internal/helm/helm.go

```diff
@@ -251,3 +256,30 @@ func GetHelmClient(restConfig *rest.Config, logger logr.Logger, namespace string
 
 	return hc.NewClientFromRestConf(options)
 }
+
+// cleanupReleaseIfNeeded will delete the helm release if it exists and is not
+// in "deployed" state. The reason is that helm will refuse to upgrade a release
+// that is in pending-install state. This would be the case, when the app container
+// is failing for whatever reason. The user may try to fix the problem by pushing
+// the application again and we want to allow that.
+func cleanupReleaseIfNeeded(l logr.Logger, c hc.Client, name string) error {
+	r, err := c.GetRelease(name)
+	if err != nil {
+		if err == helmdriver.ErrReleaseNotFound {
+			return nil
+		}
+		return errors.Wrap(err, "getting the helm release")
+	}
+
+	if r.Info.Status != helmrelease.StatusDeployed {
```

Comment:
This is the part I'm not sure about. There is no "sure" way to force a re-deployment of a helm release. Some states of the release won't let helm deploy on top no matter what (e.g. "pending install"). In those cases, one has to remove the release and deploy again (like we do here). I chose to only keep the release if it is in "deployed" state. I wonder if the history of the release is relevant in any way and be removing it we loose something. Currently we don't seem to make any use of the history.